### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by bergos**
- better separation of operations and connected code
- more readable code using clownface to extract data from datasets
- separate store for data and dataset for the API

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const { Router } = require('express')
+const rdf = { ...require('@rdfjs/data-model'), ...require('@rdfjs/dataset') }
+const rdfHandler = require('@rdfjs/express-handler')
+const setLink = require('set-link')
+const apiHeader = require('./lib/middleware/apiHeader')
+const iriTemplate = require('./lib/middleware/iriTemplate')
+const operation = require('./lib/middleware/operation')
+const resource = require('./lib/middleware/resource')
+
+function middleware (api, store) {
```
Comment (tpluscode) [inline middleware.js]: So the `store` is a combination of RDF/JS [`Store`](http://rdf.js.org/stream-spec/#store-interface) and [`Source`](http://rdf.js.org/stream-spec/#source-interface)

File: README.md [hunk 2]
```diff
@@ -2,135 +2,75 @@
...
+For `GET` requests with a matching IRI Template, the `.dataset()` and `.quadStream()` as defined by `express-handler` are also available to read the given variables.
+Additionally there is a `hydra` property assigned to `req` that contains more data about the request: 
+
+```javascript
+  req.hydra = {
+    // Api object given as argument to the middleware
+    api,
+ 
+    // RDF/JS Store object given as argument to the middleware
+    store,
```
Comment (tpluscode) [inline README.md]: I just figured that maybe the name `store` is a bit technical and not descriptive.

Common term in API space is **state** or **resource state**. So maybe something like `stateStore`? Or `resourceStateStore` but rather long...

File: Api.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    addAll(this.dataset, await fromStream(rdf.dataset(), fromFile(filePath)))
+
+    return this
+  }
+
+  replaceIRI (oldIRI, newIRI) {
+    this.dataset = replaceDatasetIRI(oldIRI, newIRI, this.dataset)
+  }
```
[the remark this answers is not in the document]
Comment (bergos) [reply Api.js]: I renamed it to just `rebase`. Besides the dataset there is nothing to rebase, so I thought the short version should also work.

File: examples/blog/api.ttl [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+<Blog> a hydra:Class;
+  hydra:search <blog#search>;
+  hydra:supportedOperation
+    <blog#get>,
+    <blog#post>.
+
+<blog#get> a hydra:SupportedOperation;
+  hydrabox:variables <blog#search>;
+  hydra:expects <TimeRange>;
```
Comment (tpluscode) [inline examples/blog/api.ttl:19]: Does `hydra:expects` have any significance here? Technically, `GET` requests would always expect `owl:Nothing`

File: examples/blog/api.ttl [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  hydra:method "GET";
+  hydra:returns <Result>;
```
Comment (tpluscode) [inline examples/blog/api.ttl:21]: This `hydra:returns` also strikes me as odd. A get method would likely return what it annotates. `Blog` in this case

File: examples/blog/blog.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    const toQuad = [...filters.match(null, ns.schema.to, null, null)][0]
+    const to = toQuad && toQuad.object
+
+    if (from || to) {
+      console.log(`filter: ${from && from.value} - ${to && to.value}`)
+    }
+  }
+
+  res.dataset(unsetGraph(req.hydra.resource.dataset))
```
Comment (tpluscode) [inline examples/blog/blog.js:23]: I wonder if this should even be necessary?

How about `req.hydra.resource.dataset` is already flattened to the default graph?

**Pull request merged**